server/controllers/admin.py

```diff
@@ -124,6 +126,17 @@ def assignment(cid, aid):
     if assgn.course != current_course:
         return abort(401)
 
+    assgn.name = form.name._value()
+    assgn.display_name = form.display_name._value()
+    assgn.due_date = datetime.strptime(form.due_date._value(), '%Y-%m-%d %H:%M:%S')       
+    assgn.lock_date = datetime.strptime(form.lock_date._value(), '%Y-%m-%d %H:%M:%S')           
```

Comment:
I don't think you have to do this if you use form.validate_on_submit() and form.populate_obj 

Also note to self: These times should be converted to UTC sometime (but that's not related to this PR)